src/tallies/MeshTally.C

```diff
@@ -176,12 +207,26 @@ MeshTally::spatialFilter()
       _libmesh_mesh_copy->allow_renumbering(false);
       _libmesh_mesh_copy->prepare_for_use();
 
-      openmc::model::meshes.emplace_back(
-          std::make_unique<openmc::LibMesh>(*_libmesh_mesh_copy.get(), _openmc_problem.scaling()));
+      if (_mesh_tally_amalgamation_post_processing)
+      {
+        openmc::model::meshes.emplace_back(std::make_unique<openmc::LibMesh>(
+            *_libmesh_mesh_copy.get(), _extra_integer_name, _openmc_problem.scaling()));
+      }
+      else
+        openmc::model::meshes.emplace_back(std::make_unique<openmc::LibMesh>(
+            *_libmesh_mesh_copy.get(), _openmc_problem.scaling()));
```

Comment:
Perhaps save this mesh reference and then only have one place to call the constructors depending on whether you have an extra integer or not.

Here it would be:
`mesh_base = *_libmesh_mesh_copy.get()`

and below it would be
`mesh_base = mesh.getMesh()`

And then you only need one copy of:
```suggestion
      if (_mesh_tally_amalgamation_post_processing)
      {
        openmc::model::meshes.emplace_back(std::make_unique<openmc::LibMesh>(
            mesh_base, _extra_integer_name, _openmc_problem.scaling()));
      }
      else
        openmc::model::meshes.emplace_back(std::make_unique<openmc::LibMesh>(
            mesh_base, _openmc_problem.scaling()));
```
also - either both of these should have {} or neither

Reply:
I defined  `auto mesh_base = *_libmesh_mesh_copy.get()` inside the if block, now getting out of the bound error which is expected. Not sure what's the best way to resolve it? Maybe I can define a `nullptr` before if else block and use that pointer to solve this issue? 

```
  libMesh::MeshBase* mesh_base_ptr;
  if (_tally_blocks.size() > 0)
  {
     //some code 
    mesh_base_ptr = _libmesh_mesh_copy.get();
  }
  else
    mesh_base_ptr = &_mesh.getMesh();

  if (_mesh_tally_amalgamation)
    openmc::model::meshes.emplace_back(std::make_unique<openmc::LibMesh>(*mesh_base_ptr, 
_clustering_name, _openmc_problem.scaling()));
  else
    openmc::model::meshes.emplace_back( std::make_unique<openmc::LibMesh>(*mesh_base_ptr, 
_openmc_problem.scaling()));
```